src/pages/commits/commitChart/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { mapTaskStatusToColor } from "constants/task";
+import { TaskStatus } from "types/task";
+
+export type GroupedResult = {
+  stats: {
+    [key: string]: { count: number; statuses: string[] };
+  };
+  max: number;
+  total: number;
+};
+
+export const groupTasksByColor = (
```

Comment:
can you write some tests for this.

Reply:
Thank you for all the comments!! They are super helpful!